### ISSUE TITLE
Catalog Search index process unknown error

### ISSUE THREAD

**Issue by Frodigo** [markshust/docker-magento#488]
**Description**
Hi, I tried to use automated setup to install Magento 2.4.2, but something is wrong with the Elastic search

**Steps To Reproduce**
1.  Run command `curl -s https://raw.githubusercontent.com/markshust/docker-magento/master/lib/onelinesetup | bash -s -- m242.test 2.4.2`
2. Console throws the error during re-indexing:
```
Catalog Search index process unknown error:
{"error":{"root_cause":[{"type":"cluster_block_exception","reason":"index [magento2_product_1_v1] blocked by: [TOO_MANY_REQUESTS/12/index read-only / allow delete (api)];"}],"type":"cluster_block_exception","reason":"index [magento2_product_1_v1] blocked by: [TOO_MANY_REQUESTS/12/index read-only / allow delete (api)];"},"status":429}
```

**Expected Result**
Magento is installed without errors,  indexer works

**Actual Result**
There is an error during re-indexing.
<img width="1412" alt="Screen Shot 2021-07-14 at 09 40 54" src="https://user-images.githubusercontent.com/11998249/125582869-b5b9d33e-315a-4ca9-9074-9ff2803e694d.png">

**Comment by Frodigo** [comment]
Running commands from this page resolves the issue: https://selleo.com/til/posts/esrgfyxjee-how-to-fix-elasticsearch-forbidden12index-read-only

**Comment by markshust** [comment]
@Frodigo so it appears these two settings are what was needed:

```
"cluster.routing.allocation.disk.threshold_enabled": false
"index.blocks.read_only_allow_delete": null
```

However, I'm a bit hesitant to pull in this changes as per https://www.elastic.co/guide/en/elasticsearch/reference/6.8/disk-allocator.html:

> Controls the flood stage watermark, which defaults to 95%. Elasticsearch enforces a read-only index block (index.blocks.read_only_allow_delete) on every index that has one or more shards allocated on the node, and that has at least one disk exceeding the flood stage. This setting is a last resort to prevent nodes from running out of disk space. The index block must be released manually when the disk utilization falls below the high watermark.

Specifically, the part about "This setting is a last resort". 

It appears this is just a low disk space issue. Is it possible your Docker daemon or host is running low on disk space?

**Comment by janssens** [comment]
I have the same issue, I still have 5Go spare disk space. I'll try this two settings.

**Comment by markshust** [comment]
@janssens thanks for the confirmation, this is a bit odd this is needed per Elasticsearch's documentation, but now that we have multiple reports I'll make this update. I'll need a few days when I circle back around and make some more updates to this repo.

**Comment by markshust** [comment]
Fixed, released and tagged in 39.0.0

**Comment by salithlalts** [comment]
Hi,
I am working with 2.3.5 version of magento..
i am getting below error when doing setup.

error: The "--elasticsearch-host" option does not exist.  

i am using below lines in install command for elasicsearch:

--elasticsearch-host=elasticsearch \
  --elasticsearch-port=9200 \
  --search-engine=elasticsearch6 \
